pypreprocess/subject_data.py

```diff
@@ -203,6 +208,26 @@ def _sanitize_session_output_dirs(self):
             self.session_output_dirs[sess] = self._sanitize_output_dir(
                 sess_output_dir)
 
+    def _sanitize_session_scratch_dirs(self):
+        """func scratch dirs (one per session)"""
+        if self.session_scratch_dirs is None:
+            if self.n_sessions is None:
+                return
+            self.session_scratch_dirs = [None] * self.n_sessions
+
+        # session-wise func scratch directories
+        for sess, sess_scratch_dir in enumerate(self.session_scratch_dirs):
+            if sess_scratch_dir is None:
+                if self.n_sessions > 1:
+                    sess_scratch_dir = os.path.join(
+                        self.scratch, self.session_ids[sess])
+                else:
+                    sess_scratch_dir = self.scratch
+            else:
+                sess_scratch_dir = self.scratch
```

Comment:
I'm not sure I understand the logic here: if `sess_scracth_dir` is not None, it gets overriden ?